scripts/leapp_upgrade.py

```diff
@@ -0,0 +1,382 @@
+            raise ProcessError(
+                message="Installation of %s (coresponding pkg to '%s') failed with exit code %s."
+                % (install_pkg, pkg, returncode)
+            )
+
+
+def remove_previous_reports():
+    print("Removing previous preupgrade reports at /var/log/leapp/leapp-report.* ...")
+
+    if os.path.exists(JSON_REPORT_PATH):
+        os.remove(JSON_REPORT_PATH)
+
+    if os.path.exists(TXT_REPORT_PATH):
+        os.remove(TXT_REPORT_PATH)
+
+
+def execute_upgrade(command):
+    print("Executing upgrade ...")
+    output, _ = run_subprocess(command)
+
+    return output
+
+
+def find_highest_report_level(entries):
+    """
+    Gather status codes from entries.
+    """
+    print("Collecting and combining report status.")
+    action_level_combined = []
+    for value in entries:
+        action_level_combined.append(value["severity"])
+
+    valid_action_levels = [
+        level for level in action_level_combined if level in STATUS_CODE
+    ]
+    valid_action_levels.sort(key=lambda status: STATUS_CODE[status], reverse=True)
+    return STATUS_CODE_NAME_MAP[valid_action_levels[0]]
+
+
+def parse_results(output):
+    print("Processing upgrade results ...")
+
+    report_json = "Not found"
+    message = "Can't open json report at " + JSON_REPORT_PATH
+    alert = True
+    status = "ERROR"
+
+    print("Reading JSON report")
+    if os.path.exists(JSON_REPORT_PATH):
+        with open(JSON_REPORT_PATH, mode="r") as handler:
+            report_json = json.load(handler)
+
+        # NOTE: with newer schema we will need to parse groups instead of flags
+        report_entries = report_json.get("entries", [])
+        inhibitor_count = len(
+            [entry for entry in report_entries if "inhibitor" in entry.get("flags")]
+        )
+        message = "Your system has %s inhibitors out of %s potential problems." % (
+            inhibitor_count,
+            len(report_entries),
+        )
+        alert = inhibitor_count > 0
+        status = (
+            find_highest_report_level(report_entries)
+            if len(report_entries) > 0
+            else "SUCCESS"
+        )
+
+    output.status = status
+    output.report_json = report_json
+    output.alert = alert
+    output.message = message
+
+    print("Reading TXT report")
+    report_txt = "Not found"
+    if os.path.exists(TXT_REPORT_PATH):
+        with open(JSON_REPORT_PATH, mode="r") as handler:
+            report_txt = handler.read()
+
+    output.report = report_txt
+
+
+def call_insights_client():
+    print("Calling insight-client in background for immediate data collection.")
+    run_subprocess(["insights-client"], wait=False)
+    # NOTE: we do not care about returncode or output because we are not waiting for process to finish
+
+
+def reboot_system():
+    print("Rebooting system in 1 minute.")
+    run_subprocess(["shutdown", "-r", "1"], wait=False)
```

Comment:
Same for shutdown here as the `insights-client` above. 


```suggestion
    run_subprocess(["/usr/sbin/shutdown", "-r", "1"], wait=False)
```